cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2562,10 +2559,16 @@ private boolean isPodToRestart(StatefulSet sts, Pod pod, boolean isAncillaryCmCh
                     if (isPodToRestart) {
                         log.debug("{}: Rolling pod {} due to {}",
                                 reconciliation, pod.getMetadata().getName(), reasons);
+                        String result = "";
+                        for (String reason: reasons) {
+                            result += "," + reason;
+                        }
+                        result = result.substring(1);
+                        return result;
```

Comment:
TBH I think it's time we refactored this method. It's maintaining two parallel pieces of state. There's the boolean logic which goes into computing `isPodToRestart` and there's the String logic around reasons. Having two parallel computations is an invitation to adding buggy code in the future. Since everything is now strings, let's get rid of the boolean logic.

Reply:
The strings are used only with `debug` logging level. Do you propose to assembly this `reasons` string every time and just use it where needed?

Reply:
Yes.